src/commands/startPage/StartPage.ts

```diff
@@ -36,6 +36,15 @@ class StartPage {
         const codiconsRoot = vscode.Uri.joinPath(ext.context.extensionUri, ...ext.ignoreBundle ? ['node_modules'] : ['dist', 'node_modules'], 'vscode-codicons', 'dist');
 
         if (!this.activePanel) {
+            const template = await this.getTemplate(resourcesRoot);

```

Comment:
With this `await`, there's a potential for reentrancy (because we haven't yet created the panel instance that would prevent this `if` branch to be taken again, though the calling function may effectively prevent this).  Maybe after the `await` test that `this.activePanel` is still `undefined`?

Reply:
Good point. I have refactored the code a bit to prevent this